doc/neps/nep-0028-deprecation_policy.rst

```diff
@@ -0,0 +1,184 @@
+currently have a place for SPEEPs (Scientific Python Ecosystem
+Enhancement Proposals) and given NumPy's central roll in our community
+this seems like the proper place to document this.
+
+
+This is being put forward by maintainers of Matplotlib, scikit-learn,
+IPython, Jupyter, yt, SciPy, NumPy, and scikit-image.
+
+
+
+Detailed description
+--------------------
+
+When making a new major or minor release, projects should support all
+minor versions of ``CPython`` initially released in the prior 42
+months from their anticipated release date and all minor version of
+NumPy released in the prior 24 months.
+
+
+The diagram::
+
+       Jan 16      Jan 17      Jan 18      Jan 19      Jan 20
+       |           |           |           |           |
+  +++++|+++++++++++|+++++++++++|+++++++++++|+++++++++++|++++++++++++
+   |              |                  |               |
+   py 3.5.0       py 3.6.0           py 3.7.0        py 3.8.0
+  |-----------------------------------------> Feb19
+            |-----------------------------------------> Dec19
+                      |-----------------------------------------> Nov20
+
+shows the support windows for ``CPython``.  A project with a minor
+release in Feb19 should support py35 and newer, a project with a minor
+release in Dec19 should support py36 and newer, and a project
+with a minor release in Nov20 should support py37 and newer.
+
+The current CPython release cadence is 18 months so a 42 month window
+ensures that there will always be at least two versions of ``CPython``
+in the window.  By padding the window by 6 months from the anticipated
+``CPython`` cadence we avoid the edge cases where a project releases
+the month after ``CPython`` and would effectively only support one
+version of ``CPython`` that has an install base (and then the window
+is 42).  This buffer also makes us resilient to minor fluctuations /
+delays in the ``CPython`` release schedule.
+
+Because the threshold for dropping a version of ``CPython`` is based
+on historical release dates and a project's plans, the decision to drop
+support for a given version of ``CPython`` can be made very early in
+the release process.
+
+There will be some unavoidable mismatch in supported version of
+``CPython`` between packages immediately after a version of
+``CPython`` ages out.  However this should not last longer than one
+release cycle of each of the projects, and when a given project
+does a minor or major release, it is guaranteed that there will be a
+stable release of all other projects that support the set of
+``CPython`` the new release will support.
+
+
+Implementation
+--------------
+
+We suggest that all projects adopt the following language into their
+development guidelines:
+
+
+   - support minor versions of ``CPython`` initially released
+     42 months prior to our planned release date
+   - always support at least the 2 latest versions of ``CPython``
```

Comment:
Does this mean Project X Version Y will support all minor version of CPython released in the 42 months prior to the release of X vY or the two most recent minor versions of CPython released before X vY, whichever is "greater"?
(I don't mean to suggest this is a clearer way of writing it; just checking understanding.)

Reply:
Yes, as commented below, this is hedging in the case that the Python release cycle slows down significantly (worst case, we don't have a Python minor release for 42 months and the rule says we support _no_ Python ;) ) .